pkg/resourcemanager/azuresql/azuresqldb/azuresqldb.go

```diff
@@ -18,6 +18,9 @@ import (
 type AzureSqlDbManager struct {
 }
 
+// Ensure we implement the interface we expect
+var _ SqlDbManager = &AzureSqlDbManager{}
```

Comment:
I'm not familiar with this syntax, what is the "_" doing? I would say its generally more go-like to use the New function than to use a global constant.

Reply:
This is a type assertion asking the compiler to ensure that we've implemented this interface. The compiler elides it at compile time (because we're assigning to a `_` variable).

It's useful in places where you have a type that impl's an interface and want a more C# (or pick your OO language of choice) interface (i.e. the type implementing the interface knows that it is doing so).

In this case, one of my previous PRs had actually changed a method signature on the AzureSqlDbManager struct but missed changing it in the SqlDbManager interface -- this is just a trick so that the compiler catches that if we make the same mistake in the future.

Reply:
it is asserting that the interface `SqlDbManager` is implemented correctly; if it isn’t then this line will fail to compile

Reply:
[snap] 🐊